### PR TITLE
dotCMS/core#26409 Calendar Design System implementation

### LINKED ISSUE #26409: Design System: Implement `p-calendar` component
### Parent
#25355

### User Story
As a dotCMS user, I want to be able to style the calendar component to match the new design system, so I can make sure my site is consistent with the new design standards.

### Acceptance Criteria

- Apply the provided Figma design to the calendar component of the primeng library.
- Ensure that the components are styled according to the provided design.
- Update the stories for Storybook

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Screenshot 2023-10-10 at 4 12 38 PM

- [Figma](https://www.figma.com/file/7CzbMcGSbTIJjerWtcapAm/Design-System-2023?type=design&node-id=807%3A11267&mode=design&t=u7HEQWwb1n7wKcop-1)
- [Primeng](https://primeng.org/calendar)

### Assumptions & Initiation Needs
- Design system guidelines are available

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates
- Update SCSS classes for calendar and all related components: 8 hours

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Implement Desing System Form Calendar CSS
* Setup Storyboard for Form/Calendar

### Screenshots
Original 
Input
Screenshot 2023-10-19 at 10 57 08 AM

Calendar
Screenshot 2023-10-19 at 10 56 48 AM

New
Screenshot 2023-10-19 at 10 53 24 AM
Screenshot 2023-10-19 at 10 53 08 AM
Screenshot 2023-10-19 at 10 46 34 AM

### <samp>🤖 Generated by Copilot at 4ea2917</samp>

This pull request updates and improves some Storybook and SCSS files in the core-web project. It refactors the Calendar story to use the new Storybook syntax and custom arguments, fixes a formatting issue in the common.scss file, adds comments to the font size variables in the fonts.scss file, and comments out some unused styles in the dot-experiments-configuration-scheduling-add.component.scss file.

## Related Issue
Fixes #26409 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 4ea2917</samp>

*  Refactor the Calendar story for Storybook, using the new args and argTypes syntax and adding custom arguments for the component's properties and controls ([link](https://github.com/dotCMS/core/pull/26469/files?diff=unified&w=0#diff-0d7fddd8dcf5fd1471dc35b4c561d706b2231469ee6fa06547bdf859e5e4f5bcL1-R83))
*  Fix a minor formatting issue in the common.scss file, adding a space before the selector for consistency and style ([link](https://github.com/dotCMS/core/pull/26469/files?diff=unified&w=0#diff-291473233533222487f2c2d6729e1c47bf788e43eeed2aab132f3123f71ce002L151-R151))
*  Add comments to the font size variables in the fonts.scss file, indicating the equivalent pixel values for reference and clarity ([link](https://github.com/dotCMS/core/pull/26469/files?diff=unified&w=0#diff-2ec369819c7145bd80f9ec0f6d11f02278eb1f0f41cbe71802e009f996f2d60cL15-R18))
*  Comment out some unused styles for the label and the calendar in the `dot-experiments-configuration-scheduling-add.component.scss` file, reducing the unnecessary CSS code and avoiding potential conflicts or overrides ([link](https://github.com/dotCMS/core/pull/26469/files?diff=unified&w=0#diff-2480f1fb45fdbe9f509329b97feefde2d0e9eba937925b73f3f6708760e9933aL6-R20))

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 1]
```diff
@@ -1,310 +1,266 @@
 @use "variables" as *;
+@import "mixins";
+@import "common";
+@import "../buttons/common";
+
+$btn-height: 2.3125rem;
+$calendar-th-height: 22px;
+$month-year-picker-width: 328px;
+$month-year-border-radius: 100px;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:9]: Why a variable to use once? 🤔

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 2]
```diff
@@ -1,310 +1,266 @@
 @use "variables" as *;
+@import "mixins";
+@import "common";
+@import "../buttons/common";
+
+$btn-height: 2.3125rem;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:6]: Not using this.

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 3]
```diff
@@ -1,310 +1,266 @@
...
+    border-radius: $border-radius-md;
+    height: $field-height-md;
+    box-sizing: border-box;
+    border: $field-border-size solid $color-palette-gray-400;
+
+    input {
+        height: auto;
+        @include truncate-text;
+    }
+    &:focus-within {
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:22]:
```suggestion

    &:focus-within {
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 4]
```diff
@@ -1,310 +1,266 @@
...
 
-.p-calendar-w-btn:not(.p-disabled):hover {
-    border-color: $black;
-}
+    .p-button:disabled.p-datepicker-trigger,
+    .p-button:enabled.p-datepicker-trigger {
+        height: auto;
+        aspect-ratio: 1/1;
+        background-color: $color-palette-gray-200;
+        .pi {
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:49]:
```suggestion

        .pi {
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 5]
```diff
@@ -1,310 +1,266 @@
...
+        &:not(.p-highlight):hover {
+            background: $color-palette-primary-100;
+        }
+    }
 }
 
-.p-datepicker .p-datepicker-header .p-datepicker-next {
-    order: 3;
-}
+p-calendar.p-calendar-clearable {
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:251]: Is this being applied?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 6]
```diff
@@ -1,310 +1,266 @@
...
 
-.p-datepicker .p-datepicker-header .p-datepicker-prev {
-    order: 2;
+    .p-monthpicker .p-monthpicker-month:not(.p-disabled),
+    .p-yearpicker .p-yearpicker-year:not(.p-disabled) {
+        &:not(.p-highlight):hover {
+            background: $color-palette-primary-100;
+        }
+    }
 }
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:249]:
```
.p-calendar .p-datepicker {
    padding: $spacing-3;
    background: $white;
    color: $black;
    border-radius: $border-radius-md;

    &:not(.p-datepicker-inline) {
        margin-top: $spacing-3;
        box-shadow: $md-shadow-6;
        min-width: auto;
        display: flex;
        flex-direction: column;
        gap: $spacing-3;
    }

    &:not(.p-disabled) {
        table td span:not(.p-highlight):not(.p-disabled) {
            &:hover {
                background: $color-palette-primary-100;
            }
        }
    
        .p-monthpicker .p-monthpicker-month:not(.p-disabled),
        .p-yearpicker .p-yearpicker-year:not(.p-disabled) {
            &:not(.p-highlight):hover {
                background: $color-palette-primary-100;
            }
        }
    }

    &.p-datepicker-timeonly .p-timepicker {
        border-top: 0 none;
    }

    .p-datepicker-group {
        display: flex;
        flex-direction: column;
        gap: $spacing-3;
    }

    .p-datepicker-group {
        .p-datepicker-header {
            color: $black;
            background: $white;
            margin: 0;
            min-height: 40px;
    
            .p-datepicker-title {
                line-height: $spacing-5;
                display: flex;
                gap: $spacing-1;
            
                .p-datepicker-year,
                .p-datepicker-month,
                .p-datepicker-decade {
                    font-size: $font-size-lg;
                    font-family: $font-default;
                    font-weight: $font-weight-regular-bold;
                    color: inherit;
            
                    &:hover {
                        color: $color-palette-primary-500;
                    }
                }
            }

            .p-datepicker-prev,
            .p-datepicker-next {
                height: $field-height-md;
                width: $field-height-md;
                border-radius: 50%;

                @extend #outlined-primary-severity;
            }
        }

        table.p-datepicker-calendar {
            tbody {
                gap: $spacing-1;
                display: flex;
                flex-direction: column;
            }
        
            tr {
                display: flex;
                gap: $spacing-1;
            }
            td,
            th {
                border-spacing: 0;
                padding: 0;
                margin: 0;
                color: $black;
            }
        
            th {
                font-size: $font-size-md;
                font-weight: $font-weight-semi-bold;
            }
        
            th > span,
            td > span {
                display: flex;
                justify-content: center;
                align-items: center;
                cursor: pointer;
            }
        
            th > span {
                width: $spacing-6;
                height: $calendar-th-height;
            }
        
            td > span {
                border-radius: 50%;
                width: $spacing-6;
                height: $spacing-6;
            }
        
            td.p-datepicker-today > span:not(.p-highlight) {
                background: $color-palette-gray-300;
                border-color: transparent;
            }
        }
    }

    .p-timepicker {
        gap: $spacing-0;
        border-top: 1px solid $color-palette-gray-300;
        padding: $spacing-3;
    
        span {
            font-size: $font-size-md;
            font-weight: $font-weight-regular-bold;
            width: $spacing-5;
            height: $spacing-5;
            display: flex;
            justify-content: center;
            align-content: center;
            flex-wrap: wrap;
    
            &.pi {
                width: $icon-lg-box;
                height: auto;
                aspect-ratio: 1/1;
                font-size: $icon-lg;
                color: $color-palette-primary-500;
            }
        }
    
        button {
            @extend #text-primary-severity;
            border-radius: $border-radius-sm;
        }
    }

    .p-monthpicker,
    .p-yearpicker {
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        justify-items: center;
        gap: $spacing-1;
        width: $month-year-picker-width;

        .p-monthpicker-month,
        .p-yearpicker-year {
            color: $black;
            width: $spacing-9;
            height: $spacing-6;
            display: flex;
            justify-content: center;
            align-items: center;
            cursor: pointer;
            transition: background-color $basic-speed;
            border-radius: $month-year-border-radius;
        }
    }

    // Days, Months and Years span container
    .p-datepicker-calendar span,
    .p-monthpicker .p-monthpicker-month,
    .p-yearpicker .p-yearpicker-year {
        &.p-disabled {
            color: $color-palette-gray-500;
        }

        &.p-highlight:not(.p-disabled) {
            // Selected
            background: $color-accessible-text-blue-bg;
            color: $color-accessible-text-blue;
        }
    }
}
```

File: core-web/apps/dotcms-ui/src/stories/primeng/form/Calendar.stories.ts [hunk 7]
```diff
@@ -1,46 +1,83 @@
...
+        showClear: true,
+        inline: false,
 
-const CalendarTemplate = `<p-calendar [showTime]="true" inputId="time" showButtonBar="true"></p-calendar>`;
-const Template: Story<Calendar> = (props: Calendar) => {
-    const template = CalendarTemplate;
+        placeholder: 'Select a date',
+        minDate: TODAY,
+        disabledDates: DISABLED_DAYS,
+        selectionMode: 'single'
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/stories/primeng/form/Calendar.stories.ts:28]: Controls are not working in the docs page.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss (hunk 5 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:251]: Yes.

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss (hunk 3 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:22]: spacing

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss (hunk 2 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:6]: No. Deleted.

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss (hunk 4 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:49]: done.

**Review by fmontes — APPROVED** [review]

**Pull request merged**